crates/bevy_ui/src/focus.rs

```diff
@@ -221,13 +221,9 @@ pub fn ui_focus_system(
         // reverse the iterator to traverse the tree from closest nodes to furthest
         .rev()
         .filter_map(|entity| {
-            let Ok(node) = node_query.get_mut(*entity) else {
-                return None;
-            };
+            let node = node_query.get_mut(*entity).ok()?;
 
-            let Some(view_visibility) = node.view_visibility else {
-                return None;
-            };
```

Comment:
this lint doesn't respect the semantic, or the intent of the code. for example it this two cases, the `return None` make it obvious when the entity if filtered, instead of it just being in the way of the standard flow of the function

Reply:
I agree here: thanks for pointing this out. The return inside of a iterator method feels like a false positive.

Reply:
I don't agree. The same argument applies to anything outside iterator method closures. An early return is important. Yet we don't reject usage of `?` in those cases.

Another question to ask is whether this would have raised concerns if the change was made the other way around: Replacing `?` by `let-else`.

To me it's obvious people would have disagreed, as it's more code to do the same thing.

Reply:
While I agree with @nicopap here, the `.ok()` rewrites aren't required by the lint and can be kept with the `return Some;`  equivalent where it makes sense.